### PR TITLE
add new test for exporting TB symbols under tf.summary

### PR DESCRIPTION
**Pull request by nfelt**
This adds a new python-based integration test that for TF 2.0, the Tensorboard-provided symbols are successfully exported under `tf.summary`.  We already have tests for this in build_pip_package.sh, but that only runs regularly in our Travis CI, whereas this python test easily can run wherever our python tests run, including our internal CI (which I verified with a test sync).

This should help avoid issues like https://github.com/tensorflow/tensorflow/issues/28027.

### CODE REVIEW

File: tensorboard/summary/tf_summary_test.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      if hasattr(tf, 'compat') and hasattr(tf.compat, 'v2'):
+        tf = tf.compat.v2
+      else:
+        self.skipTest('TF v2 summary API not available')
+    # Check that tf.summary contains both TB-provided and TF-provided symbols.
+    expected_symbols = frozenset(
+        ['scalar', 'image', 'audio', 'histogram', 'text']
+        + ['write', 'create_file_writer', 'SummaryWriter'])
+    self.assertLessEqual(expected_symbols, frozenset(dir(tf.summary)))
+    # Ensure we can deference symbols as well.
```
Comment (wchargin) [inline tensorboard/summary/tf_summary_test.py]: sp.: *deference* → *dereference*

File: tensorboard/summary/tf_summary_test.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import collections
+import sys
+import unittest
+
+class TfSummaryExportTest(unittest.TestCase):
+
+  def test_tf_summary_export(self):
+    # Check as a precondition that TF wasn't already imported.
+    self.assertEqual('notfound', sys.modules.get('tensorflow', 'notfound'))
```
Comment (wchargin) [inline tensorboard/summary/tf_summary_test.py]: Why is this needed, given that we immediately import TensorFlow?

The existing occurrence of this assertion in `summary_test.py` ensures
that TensorFlow isn’t automatically imported by accessing certain
symbols, but I don’t see why we need something like that here.

File: tensorboard/summary/tf_summary_test.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+We also test this at an even higher level (and with many different import
+sequences) via our pip package build script. But that script is run only in
+our external CI, whereas this python test can run anywhere our normal tests run.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
```
Comment (wchargin) [inline tensorboard/summary/tf_summary_test.py]: Unused import.

File: tensorboard/summary/tf_summary_test.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import sys
+import unittest
+
```
Comment (wchargin) [inline tensorboard/summary/tf_summary_test.py]: nit: Add blank line.

File: tensorboard/summary/tf_summary_test.py (hunk 3 above)
Comment (nfelt) [reply tensorboard/summary/tf_summary_test.py]: Done.

File: tensorboard/summary/tf_summary_test.py (hunk 4 above)
Comment (nfelt) [reply tensorboard/summary/tf_summary_test.py]: Done.

File: tensorboard/summary/tf_summary_test.py (hunk 2 above)
Comment (nfelt) [reply tensorboard/summary/tf_summary_test.py]: I think this is a stronger test if we ensure that it's testing the entire flow of importing TF and then using tf.summary, versus having the import happen "somewhere else".  This check enforces that we don't modify this test down the road to import TF at the top of the file or in another test method, which would muddy state going into this test.

I've elaborated the comment to explain this reasoning better.

File: tensorboard/summary/tf_summary_test.py (hunk 1 above)
Comment (nfelt) [reply tensorboard/summary/tf_summary_test.py]: Done.

File: tensorboard/summary/tf_summary_test.py (hunk 2 above)
Comment (wchargin) [reply tensorboard/summary/tf_summary_test.py]: Okay, fine with me. Thanks.
